### DISCUSSION TITLE
HELP! S7 Communication Error?

### DISCUSSION THREAD

**Discussion by Halyeah** [artisan-roaster-scope/artisan#625] [Q&A] [answered]
I'm having problems connecting my Macbook to my Giesen W15, please help? 

![IMG_9495 2](https://user-images.githubusercontent.com/85431669/120917005-3b0b9e80-c6f0-11eb-8a1a-aed44de80249.JPG)

**Comment by MAKOMO** [comment]
Please check the [Supported Machine page for Giesen ](https://artisan-scope.org/machines/giesen/) which gives hints like "The computer running Artisan must be on the same IP network as the Giesen roaster.".

Obviously your PC is not configured to be on the same IP network than your Giesen machine and thus cannot communicate to it.

**Comment by Halyeah** [comment]
Thanks Artisan-Roaster-Scope/Artisan,

I'm a Coffee guy not a tech guy, so thanks for the help and I apologies for
having no understanding of networks.

Thanks,
Lucas

**Comment by keikoffie** [comment]
Hi, was the problem solved? I have a simulair problem unfortanly. I use phidgeds and have a FIME italian coffee roaster. Exact the same fault and looking for answers

**Comment by keikoffie** [comment]
![Uploading IMG_20210721_112715048.jpg…]()

**Reply by MAKOMO** [reply]
So you do not have a Giesen and you do not connect to your roaster via the network, but you connect to your Phidgets via USB? Still you get S7 errors? Well, you might have a mixed Artisan setup that still defines some S7 device setups and those throw errors as they try to connect to something you do not have. Maybe it is best to start from a clean Artisan setup by using the factory reset under the Help menu and then configure for your Phidget setup.

**Reply by keikoffie** [reply]
Hi, yes it's with phidgeds with the USB kabel. This afternoon i did do the factory settings but unfortanly no differents. Tomorrow morning wil try again 🙂

**Reply by keikoffie** [reply]
Hi, Just starter up with new factory settings. I don't have errrors but only tekst scope monitoring... I see no bt and et unfortanly. What can be the problem?

**Comment by Halyeah** [comment]
Hi Marko,

connection error continue.
This is the message I get on Artisan
S7 communication Error readlnt (5,2,16, False):b’ TCP: Network is
***@***.***

My roaster is a Giesen W15, with unlock code, I have been using artisan
with this toaster for a couple of years.  I get this s7 communication error
regularly, today I can not get my Mac to connect to roaster
I have checked IP Address swapped laptops, downloaded artisan again, I end
up with the same result.

Please, is there anything you can suggest to solve?

**Reply by poundy** [reply]
When you say you "checked IP address" what did you actually do? Do you know how to test that you can communicate to the IP address of the roaster outside of Artisan? To me, as Marko pointed out in the first post, the error states quite simply that the TCP/IP address you're trying to communicate with is not reachable. Given you say you see this often, that leads me to think there's one of two issues going on here, either your network is changing and causing you issues, or there's a problem connecting to your network reliably (router or switch causing issues).  Perhaps you could state how the network is configured, what cables you use etc, so we can form a better picture there?

**Comment by MAKOMO** [comment]
Let me ask one more question: is this a temporary situation that happens sometimes after some roasts or is this is a stable state and you cannot connect anymore at all?

**Comment by Knightey** [comment]
Hi, 

I am facing a similar issue on my Kuban Supreme roaster with PLC  communication: 
"**S7 Communication Error readFloat(5,94,26,False): [Errno 24] Too many open files? 'nul'@line 516**"
![S7 communication Error 2](https://user-images.githubusercontent.com/96001438/145707795-1bb5dc7c-5475-400f-b21b-31adc853034c.JPG)

This issue has something to do with the PLC code or with Artisan SW? 
The issue is random for now.

Any idea?

**Reply by MAKOMO** [reply]
Which version of Artisan on which OS are you using?

**Reply by Knightey** [reply]
My Artisan is all the time with the latest updates. 
I am running the Artisan on MS Windows 10 64b.

**Comment by MAKOMO** [comment]
Could you give the current continuous build a try and report if this performs any better?

https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

Thanks!

**Reply by Knightey** [reply]
Hello Marko,

today I have tried this version 2.4.7 but it looks quite different (fonts, graphs....) and it is buggy (doesn't react on the sliders if you moved by mouse only by one or two % steps. If you click by keyboard UP/Down, then it is OK). After one roast I have to uninstall it and install back the stable 2.4.6 version. Also, the Analyzer gives you different values. 

Btw: a week ago I still got another S7 communication error: 
"S7 Communication Error writeBool(5,94,26,False): [Errno 24] Too many open files? 'nul'@line 491"

**Reply by MAKOMO** [reply]
Knightey, your report lacks all details to track things down. Could you please attach here your artisan-settings.aset file as exported from menu Help >> Save Settings. Could you please tell us on which OS you are running (macOS, Windows, Linux)? Could you drag-and-drop here a picture of the font/graph issues you see under 2.4.7 and how that looks different on 2.4.6? Could you please be more specific on what does not work exactly with the sliders? Maybe attach a small video demonstrating the fact?

Thanks for your support,
Marko

**Reply by MAKOMO** [reply]
All those S7 communication errors are the same. Your operating system does not allow to open another pipe to send data to the machine. Most likely as previously opened pipes were not closed correctly on error.

**Comment by Halyeah** [comment]
Could you explain further, if previously open pipes are not closed
correctly on error, how do I correct this?
I am still trying to understand why I have irregular communication errors,
I don't change set up or settings from week to week, then randomly I can
not connect and waste hours trying to resolve the issue.

**Comment by MAKOMO** [comment]
You cannot fix this. Which Artisan version are you using?

You can try reducing the number of requests send by disabling oversampling in the Config menu. You should also ensure to have "optimize" ticked in menu Config >> Ports, MODBUS tab.

Otherwise, you could test (and report!) if our development version for the next release operates and better:

https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

**Comment by Halyeah** [comment]
I have been using Version 2.4.4
Oversampling already disabled and Optimize already selected in the modbus
configure.

This problem occurs about every 2 months, Artisan connects and works great
week in week out, then for reasons I don't understand 'S7 Connection error'

I'll try the latest version

Thanks again for help

**Comment by MAKOMO** [comment, marked as answer]
We might have found and resolved this problem finally (Issue #816). The fix will be part of the next Artisan release.

**Comment by kentchiong94** [comment]
![IMG_1982](https://github.com/artisan-roaster-scope/artisan/assets/169187253/3efbc8e5-1bb3-4599-95cc-26cb28e20661)


Hi i hope you can help me with this issue

**Reply by poundy** [reply]
First, we know nothing about your setup or what you think has gone wrong here. You've posted it in someone else's problem on a Giesen W15 from a macbook - is that the same situation you have? If not, you'd be better off starting your own thread. And providing more detailed information would be useful and smart - obvious things like have you ever had Artisan working against this roaster before and you've only just started having this issue on upgrade to 2.10.4, or are you trying to do a first-time setup... You should likely export your settings out to a file, rename that file with a .txt extension, then drag and drop that into the guthub reply field so it properly attaches (ie don't use email, they get stripped)

**Reply by MAKOMO** [reply]
The error message is indicating that Artisan could not reach out to the Giesen at all. The reason is a network issue. Either the PC running Artisan is not in the correct network to make the Giesen accessible or Artisan is configured with the wrong IP address for the Giesen. Please open a separate documentation item for this. For sure this is a pure configuration issue on either the network/PC side or the IP address specified in the S7 tab on Artisan, and not a software issue.

**Comment by SiebenBergeRheinland** [comment]
Hello everyone, I also have the problem that I get the S7 error message. I didn't have this problem last week! I work with a MacBook and a W15 from 2020. Last week I was able to access the W15 via the browser, but that doesn't work anymore. On the MacBook, do I have to enter the same IP as the roaster, with the last two numbers changed to 42? For Artisan, I enter 172,30,50,50 as the IP and on the MacBook, 172,30,50,42. Have I understood everything correctly? Thank you for your help.

**Reply by poundy** [reply]
which error, the one in the main post for this thread, or the one immediately above posted by kentchiong94, as they are different and mean different things (but may point to the same fundamental issue).
You need to correctly match the network information posted in https://artisan-scope.org/machines/giesen/.  Can you clear up what changed between last week (when it was working) and this week (when it is broken)? Did you reconfigure anything, did you change the Mac or the software related to it, did you change network settings, or did you change settings in Artisan? The requirement is that the IP addresses are on the same network as each other. Depending on the year of your machine, and the hardware you have, you will need to match the list given in the link above. The 172.30.50.42 IP address is an EXAMPLE IP address to set on your PC/Mac/Device running Artisan, so it can communicate to a roaster manufactured after 2/2018. You can use the last number (the .42) to be any number that doesn't clash with the allocated number on the roaster, but the first three octets need to be an exact match, and your subnet mask needs to be 255.255.255.0.

**Comment by SiebenBergeRheinland** [comment]
Hello, thanks for the answer. I honestly don't know what has changed in the setup since last week, I don't think I've changed anything. I get this message:

**Comment by SiebenBergeRheinland** [comment]
<img width="820" alt="Bildschirmfoto 2024-09-25 um 10 23 56" src="https://github.com/user-attachments/assets/2e3ba886-d64f-4cce-b823-50b98475263a">

**Comment by SiebenBergeRheinland** [comment]
I have already connected the machine.
<img width="895" alt="Bildschirmfoto 2024-09-25 um 10 25 04" src="https://github.com/user-attachments/assets/d0faa9e1-aaa9-4eb6-bd32-515efb227a6a">
<img width="378" alt="Bildschirmfoto 2024-09-25 um 10 25 34" src="https://github.com/user-attachments/assets/d8a2caf9-e6dd-43a7-a5a2-183c44a36cf4">

**Comment by SiebenBergeRheinland** [comment]
<img width="568" alt="Bildschirmfoto 2024-09-25 um 10 26 09" src="https://github.com/user-attachments/assets/0c94227b-1615-4847-96fa-e098b66099f1">

**Reply by MAKOMO** [reply]
Is the macOS Firewall active?

**Comment by SiebenBergeRheinland** [comment]
hello, we found the problem, it was the wrong IP address. Instead of 172.30.30.50, the machine has the IP address http://192.168.2.180. The seller told me the production date of the machine was 2021, but that doesn't match the IP address

**Reply by poundy** [reply]
I am surprised then that it was working for you last week - but you didn't know the address. Please look up how to save your settings to a file and keep that as backup in case you ever have this kind of thing happen again

**Reply by MAKOMO** [reply]
There was an additional problem. The Artisan app did not had the permission to access the local network. This can be fixed in macOS under System Settings, Privacy & Security, Local Network. There enable the Artisan apps access.
